src/gnssro/gnssro_bufr2ioda.f90

```diff
@@ -62,39 +60,47 @@ program gnssro_bufr2ioda
       integer(i_kind), allocatable, dimension(:)    :: recn
       integer(i_kind), allocatable, dimension(:)    :: asce
       integer(i_kind), allocatable, dimension(:)    :: ogce
-      real(r_double), allocatable, dimension(:)     :: time
-      character(len=20), allocatable, dimension(:)  :: datetime
-      real(r_double), allocatable, dimension(:)     :: lat
-      real(r_double), allocatable, dimension(:)     :: lon
-      real(r_double), allocatable, dimension(:)     :: rfict
-      real(r_double), allocatable, dimension(:)     :: azim
-      real(r_double), allocatable, dimension(:)     :: geoid
-      real(r_double), allocatable, dimension(:)     :: msl_alt
-      real(r_double), allocatable, dimension(:)     :: ref
-      real(r_double), allocatable, dimension(:)     :: refoe_gsi
-      real(r_double), allocatable, dimension(:)     :: bend_ang
-      real(r_double), allocatable, dimension(:)     :: impact_para
-      real(r_double), allocatable, dimension(:)     :: bndoe_gsi
+      real(r_kind), allocatable, dimension(:)       :: time
+      integer(i_64),allocatable, dimension(:)     :: epochtime
+      real(r_kind), allocatable, dimension(:)     :: lat
+      real(r_kind), allocatable, dimension(:)     :: lon
+      real(r_kind), allocatable, dimension(:)     :: rfict
+      real(r_kind), allocatable, dimension(:)     :: azim
+      real(r_kind), allocatable, dimension(:)     :: geoid
+      real(r_kind), allocatable, dimension(:)     :: msl_alt
+      real(r_kind), allocatable, dimension(:)     :: ref
+      real(r_kind), allocatable, dimension(:)     :: refoe_gsi
+      real(r_kind), allocatable, dimension(:)     :: bend_ang
+      real(r_kind), allocatable, dimension(:)     :: impact_para
+      real(r_kind), allocatable, dimension(:)     :: bndoe_gsi
 end type gpsro_type
 
 type(gpsro_type) :: gpsro_data
 
-real(r_double),dimension(n1ahdr)     :: bfr1ahdr
-real(r_double),dimension(50,maxlevs) :: data1b
-real(r_double),dimension(50,maxlevs) :: data2a
-real(r_double),dimension(maxlevs)  :: nreps_this_ROSEQ2
-integer(i_kind)           :: iret,levs,levsr,nreps_ROSEQ1,nreps_ROSEQ2_int
-real(r_double) :: pcc,qfro,usage,dlat,dlat_earth,dlon,dlon_earth,freq_chk,freq,azim
-real(r_double) :: height,rlat,rlon,ref,bend,impact,roc,geoid,  bend_error,ref_error,bend_pccf,ref_pccf
-real(r_double) :: obsErr
-real(r_double),   parameter :: missingvalue=-9.9e10
+real(r_kind),dimension(n1ahdr)     :: bfr1ahdr
+real(r_kind),dimension(50,maxlevs) :: data1b
+real(r_kind),dimension(50,maxlevs) :: data2a
+real(r_kind),dimension(maxlevs)    :: nreps_this_ROSEQ2
+integer(i_kind)                    :: iret,levs,levsr,nreps_ROSEQ1,nreps_ROSEQ2_int
+real(r_kind) :: pcc,qfro,usage,dlat,dlat_earth,dlon,dlon_earth,freq_chk,freq,azim
```

Comment:
qfro can be declared as qfro(1) to eliminate a gnu compilation warning.

Reply:
Please, see the comment above.